uast/uast.go

```diff
@@ -262,87 +270,311 @@ type Any interface{}
+	// - Alias (contains any of the above and defines a local package name within a file/scope);
+	Path Any `json:"Path"`
+
+	// All is set to true when the statement defines all exported symbols from
+	// a module in the local scope (usually file).
 	All    bool  `json:"All"`
 	Names  []Any `json:"Names"`
 	Target Scope `json:"Target"`
 }
 
+// RuntimeImport is a type of an import statement that imports a module only when an execution
+// reaches this UAST node. The import side effects are executed only once, regardless of how many
+// times a statement is reached.
+//
+// Similar to import in PHP, Python, JS.
+//
+// For other import types, see Import.
 type RuntimeImport Import
 
+// RuntimeReImport is a subset of RuntimeImport statement that will re-execute an import and its
+// side-effects statement each time an execution reaches the statement.
+//
+// Similar to import in PHP, Python.
+//
+// For other import types, see Import.
 type RuntimeReImport RuntimeImport
 
+// InlineImport is a subset of import statement that acts like a preprocessor - all statements in
+// the imported module are copied into a position of the UAST node.
+//
+// Similar to import in C and C++.
+//
+// For other import types, see Import.
 type InlineImport Import
 
+// Argument is a named argument or return of a function.
 type Argument struct {
 	GenNode
-	Name        *Identifier `json:"Name"`
-	Type        Any         `json:"Type"`
-	Init        Any         `json:"Init"`
-	Variadic    bool        `json:"Variadic"`
-	MapVariadic bool        `json:"MapVariadic"`
-	Receiver    bool        `json:"Receiver"`
+	// Name is an optional name of an argument.
+	Name *Identifier `json:"Name"`
+
+	// Type is an optional type of an argument.
+	Type Any `json:"Type"`
+
+	// Init is an expression that should be used to initialize an argument
+	// in case no value is provided. Also known as a default value of an argument.
+	Init Any `json:"Init"`
+
+	// Variadic flag may be set only for the last argument to denote a function
+	// with a variadic number of arguments.
+	Variadic bool `json:"Variadic"`
+
+	// MapVariadic flag may be set only for the last argument to denote a function
+	// that accepts a map/dictionary value that is mapped to function arguments.
+	MapVariadic bool `json:"MapVariadic"`
+
+	// Receiver is set to true if an argument is a receiver of a method call.
+	Receiver bool `json:"Receiver"`
 }
 
+// FunctionType is a signature of a function.
 type FunctionType struct {
 	GenNode
+	// Arguments is a set of arguments the function accepts.
+	//
+	// Methods defined on structures and classes should have the first argument
+	// that corresponds to a method's receiver ("this" in most languages).
 	Arguments []Argument `json:"Arguments"`
-	Returns   []Argument `json:"Returns"`
+
+	// Returns is a set of values returned by a function.
+	//
+	// Languages with an implicit return should specify a single return with an
```

Comment:
Does this imply that if `len(a.Returns) == 0` that it's explicitly a procedure returning no values?